array.c

```diff
@@ -5067,32 +5067,220 @@ rb_ary_clear(VALUE ary)
 
 /*
  *  call-seq:
- *     ary.fill(obj)                                 -> ary
- *     ary.fill(obj, start [, length])               -> ary
- *     ary.fill(obj, range)                          -> ary
- *     ary.fill {|index| block}                      -> ary
- *     ary.fill(start [, length]) {|index| block}    -> ary
- *     ary.fill(range) {|index| block}               -> ary
+ *    array.fill(obj) -> self
+ *    array.fill(obj, start) -> self
+ *    array.fill(obj, start , length) -> self
+ *    array.fill(obj, range) -> self
+ *    array.fill { |index| ... } -> self
+ *    array.fill(start) { |index| ... } -> self
+ *    array.fill(start , length) { |index| ... } -> self
+ *    array.fill(range) { |index| ... } -> self
+ *
+ *  Replaces specified elements in +self+ with specified objects; returns +self+.
+ *
+ *  - Arguments +start+ and +length+, if given, must be
+ *    {Integer-convertible objects}[doc/implicit_conversion_rdoc.html#label-Integer-Convertible+Objects].
+ *  - Argument +range+, if given, must be a \Range object.
+ *
+ *  ---
+ *
+ *  With argument +obj+ and no block given, replaces all elements with that one object:
+ *    a = ['a', 'b', 'c', 'd']
+ *    a # => ["a", "b", "c", "d"]
+ *    a1 = a.fill(:X)
+ *    a1 # => [:X, :X, :X, :X]
+ *    a.equal?(a) #  => true # Retrurned self
+ *
+ *  ---
+ *
+ *  With arguments +obj+ and +start*, and no block given, replaces elements based on the given start.
```

Comment:
This `+` swapped itself for a `*`
```suggestion
 *  With arguments +obj+ and +start+, and no block given, replaces elements based on the given start.
```